zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -672,6 +685,14 @@ public Stat setData(String path, byte[] data, int version, long zxid, long time)
             n.stat.setVersion(version);
             n.copyStat(s);
             nodes.postChange(path, n);
+            EphemeralType ephemeralType = EphemeralType.get(n.stat.getEphemeralOwner());
+            if (ephemeralType == EphemeralType.TTL) {
+                TTLNode ttlNode = ttlMap.get(path);
```

Comment:
This `get` is not needed. Just remove from `ttlMap` and create a new `TTLNode` to add to `ttls` and `ttlMap`. Then, `expireTime` in `TTLNode` can be made final and the setter removed.